### ISSUE TITLE
Source Obsolete attribute with array named argument doesn't produce diagnostics

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#42771]
**Version Used**: 441c154

**Steps to Reproduce**:

```cs
using System;
#pragma warning disable 436

namespace System
{
    public class ObsoleteAttribute : Attribute
    {
        public int[] P1 { get; set; }
    }
}

class C1
{
    [Obsolete(
        P1 = new[] { 0 }
    )]
    void M1() { }
    
    void M2()
    {
        M1(); // expect an obsolete diagnostic here but get none
    }
}
```

**Expected Behavior**:
Obsolete diagnostic is issued for call to `M1()`.

**Actual Behavior**:
No diagnostic is issued.

- If the attribute is modified to not pass an argument for `P1`, the diagnostic is issued on usage.
- If the obsoleted member is compiled and referenced from metadata, the diagnostic is issued on usage.
